### PR TITLE
test: Code Coverage: Add Edit Mileage #3 Part 1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at e04ac6f</samp>

Refactored and tested the `AddEditMileagePage` component to handle optional fields and improve error handling. Split the unit tests into two files: `add-edit-mileage-1.spec.ts` and `add-edit-mileage-2.spec.ts` for better readability and maintainability.

### <samp>🤖 Generated by Copilot at e04ac6f</samp>

> _Sing, O Muse, of the skillful coder who refines the mileage form_
> _And splits the tests in two, to ease the burden of his craft_
> _`AddEditMileagePage`, the well-designed component, he transforms_
> _With optional fields and error checks, and cleanses of the chaff_

### <samp>🤖 Generated by Copilot at e04ac6f</samp>

*  Add unit tests for custom date validator, getEditExpense, and checkAvailableAdvance functions in `add-edit-mileage-1.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R884-R1052))
*  Create `add-edit-mileage-2.spec.ts` to contain the second set of unit tests for `AddEditMileagePage` component ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-035f26539504516a07a4540ded16b687e8c0447825f17da32e9cfc9c47c9e3f5R1-R185))
*  Change return type of `getFormValues` method in `AddEditMileagePage` class to `Partial<FormValue>` to allow for optional properties ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL358-R358))
*  Modify `getFormValues` method call in `AddEditMileagePage` class to use optional chaining operators to avoid errors ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL2193-R2182))
*  Remove unused `customDistanceValidator` method in `AddEditMileagePage` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL946-L956))
*  Remove unused `orgCategoryData` import in `add-edit-mileage-1.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961L13-R13))
*  Add `FormControl` import in `add-edit-mileage-1.spec.ts` to use in custom date validator tests ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961L3-R3))
*  Reorder and group imports in `add-edit-mileage-1.spec.ts` alphabetically and by source ([link](https://github.com/fylein/fyle-mobile-app/pull/2269/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961L58-R69))

## Clickup
https://app.clickup.com/t/85ztjmtf6

## Code Coverage
`22.89% Statements 193/843`
`17.32% Branches 119/687`
`17.68% Functions 52/294`
`23.78% Lines 191/803`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts [hunk 1]
```diff
@@ -879,5 +881,174 @@ export function TestCases1(getTestBed) {
...
+      it('should return null when the date is within the valid range', () => {
+        const control = new FormControl('2023-06-15');
+        const result = component.customDateValidator(control);
+        expect(result).toBeNull();
+      });
+
+      it('should return an error object when the date is after the upper bound of the valid range', () => {
+        const tomorrow = new Date();
+        tomorrow.setDate(tomorrow.getDate() + 2);
+        const control = new FormControl(tomorrow.toISOString().substring(0, 10));
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts:895]: is this to get the date string specifically? if yes, can we use tomorrow.toDateString()?

File: src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts [hunk 2]
```diff
@@ -879,5 +881,174 @@ export function TestCases1(getTestBed) {
...
+        const result = component.customDateValidator(control);
+        expect(result).toEqual({ invalidDateSelection: true });
+      });
+    });
+
+    it('getEditExpense(): should return an unflattened expense to edit', (done) => {
+      transactionService.getETxnUnflattened.and.returnValue(of(unflattenedTxnData));
+      activatedRoute.snapshot.params.id = JSON.parse(JSON.stringify(unflattenedTxnData.tx.id));
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-1.spec.ts:903]: why do we need JSON.parse and JSON.stringify here?

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 3]
```diff
@@ -943,17 +943,6 @@ export class AddEditMileagePage implements OnInit {
     }
   }
 
-  customDistanceValidator(control: AbstractControl): null | { invalidDistance: boolean } {
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:946]: Was this unused?

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 3 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:946]: Yes, this function was not used anywhere

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**